Fix case sensitivity with vote names and add "mode" option

rootatdebian opened the pull request:
This fixes case sensitivity with player vote names and adds a "mode" option to the rewards config. The "mode" option defaults to "random", so no sudden unexpected behaviour will occur upon updating the plugin.

rootatdebian committed:
Add new "mode" config option

Add a new config option called "mode" inside the rewards configuration file. When set to "random", the player will be given a number of random rewards from the list specified by the "amount" option. When set to "all", the player will be given one of every reward in the list.

rootatdebian committed:
Make player vote name case insensitive

hsyyid commented:
You need to add the default value for this in the `RewardsConfig` class, in one of the methods - don't recall which one right now, but you should see it.

rootatdebian committed:
Make the config populate the "mode" option

If the rewards config file doesn't exist, populate the "mode" option with its default value of "random".

rootatdebian committed:
Set the "mode" option to default if invalid

If the "mode" option in the rewards config file is not set to "all" or
"random" (valid values), set it to a valid default value of "random".

rootatdebian commented:
These commits have not been tested. The build did pass, however.

hsyyid commented:
Okay, I'll test it and merge in a bit

rootatdebian commented:
You can now close the open issue for case insensitivity. Also, you might want to make release v0.7; unless, of course, you're planning more features for it.
